build/Utils.cmake

```diff
@@ -235,7 +235,7 @@ function(resolve_buck2)
 
   set(resolve_buck2_command
       ${PYTHON_EXECUTABLE} ${executorch_root}/build/resolve_buck.py
-      --cache_dir=${CMAKE_CURRENT_BINARY_DIR}/buck2-bin
```

Comment:
Move the buck2-bin to root dir, so that it can be cleaned up? Isn't already covered by cleaning up `CMAKE_CURRENT_BINARY_DIR` (i.e. `cmake-out`)?

Reply:
Yeah, did that here

https://github.com/pytorch/executorch/pull/7305